test/gtest/bad_fusion_plan.cpp

```diff
@@ -0,0 +1,211 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <random>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/solver_id.hpp>
+#include <serialize.hpp>
+#include <fusionHost.hpp>
+#include <miopen/fusion.hpp>
+#include <miopen/fusion/solvers.hpp>
+#include <miopen/fusion/fusion_invoke_params.hpp>
+
+#include "tensor_util.hpp"
+#include "get_handle.hpp"
+
+template <typename T>
+miopenDataType_t GetDataType();
+
+template <>
+miopenDataType_t GetDataType<half_float::half>()
+{
+    return miopenHalf;
+}
+
+struct ConvTestCase
+{
+    size_t N;
+    size_t C;
+    size_t H;
+    size_t W;
+    size_t k;
+    size_t y;
+    size_t x;
+    size_t pad_x;
+    size_t pad_y;
+    size_t stride_x;
+    size_t stride_y;
+    size_t dilation_x;
+    size_t dilation_y;
+    friend std::ostream& operator<<(std::ostream& os, const ConvTestCase& tc)
+    {
+        return os << "(N: " << tc.N << " C:" << tc.C << " H:" << tc.H << " W:" << tc.W
+                  << " k: " << tc.k << " y:" << tc.y << " x:" << tc.x << " pad_y:" << tc.pad_y
+                  << " pad_x:" << tc.pad_x << " stride_y:" << tc.stride_y
+                  << " stride_x:" << tc.stride_x << " dilation_y:" << tc.dilation_y
+                  << " dilation_x:" << tc.dilation_x << " )";
+    }
+    std::vector<size_t> GetInput() const { return {N, C, H, W}; }
+    std::vector<size_t> GetWeights() const { return {k, C, y, x}; }
+    miopen::ConvolutionDescriptor GetConv() const
+    {
+        return miopen::ConvolutionDescriptor{
+            {static_cast<int>(pad_y), static_cast<int>(pad_x)},
+            {static_cast<int>(stride_y), static_cast<int>(stride_x)},
+            {static_cast<int>(dilation_y), static_cast<int>(dilation_x)}};
+    }
+};
+
+const static ConvTestCase conv_config = {64, 64, 56, 56, 64, 3, 3, 1, 1, 1, 1, 1, 1};
+
+template <typename Solver, typename T>
+class TestFusionPlan
+{
+
+public:
+    TestFusionPlan(const miopenTensorLayout_t& tensor_layout,
+                   const miopenActivationMode_t& activ_mode)
+        : handle(get_handle())
+    {
+
+        input   = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetInput()};
+        weights = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetWeights()};
```

Comment:
If we are just checking applicability then we dont need the full tensor, just the descriptor